WPFHospitalEditor/SearchResultDialog.xaml.cs

```diff
@@ -287,9 +296,12 @@ private List<MapObject> getObjects(String building, String floor)
 
         private bool isBuildingAndFloorEqual(String building, String floor, MapObject mapObjectForChecking)
         {
-            if (building.Equals(getBuildingAndFloor(mapObjectForChecking).Item1) && floor.Equals(getBuildingAndFloor(mapObjectForChecking).Item2))
+            if (getBuildingAndFloor(mapObjectForChecking) != null)
             {
-                return true;
+                if (building.Equals(getBuildingAndFloor(mapObjectForChecking).Item1) && floor.Equals(getBuildingAndFloor(mapObjectForChecking).Item2))
+                {
+                    return true;
+                }
```

Comment:
it is cleaner if you call the getBuildingAndFloor function once and put the return value in a variable. Then you just use that variable in the rest of the code.

Reply:
Good point, app runs faster too.